File: sample_onsr/oci-objectstorage-onsr-put-object-python/func.py
#
# oci-objectstorage-onsr-put-object-python version 1.0.
#
# Copyright (c) 2020 Oracle, Inc.
# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
#

import io
import os
import json
import sys
from fdk import response

import oci.object_storage

# Certs are mounted at this location for ONSR realms
cert_file = "/python/certifi/cacert.pem"
# file to upload
file_to_upload = "onsr_cert_test"
file_to_upload_content = {"content":"This is test file for ONSR test"}

def handler(ctx, data: io.BytesIO=None):
    try:
        body = json.loads(data.getvalue())
        bucketName = body["bucketName"]
    except Exception:
        error = """
                Input a JSON object in the format: '{"bucketName": "<bucket name>",
                "content": "<content>", "objectName": "<object name>"}'
                """
        raise Exception(error)
    signer = oci.auth.signers.get_resource_principals_signer()
    client = oci.object_storage.ObjectStorageClient(config={}, signer=signer)
    #client.base_client.session.cert = cert_file

    resp = put_object(client, bucketName, file_to_upload, file_to_upload_content)
    return response.Response(
        ctx,
        response_data=json.dumps(resp),
        headers={"Content-Type": "application/json"}
    )

def put_object(client, bucketName, objectName, content):
    namespace = client.get_namespace().data
    output=""
    try:
        object = client.put_object(namespace, bucketName, objectName, json.dumps(content))
        output = "Success: Put object '" + objectName + "' in bucket '" + bucketName + "'"
    except Exception as e:
        output = "Failed: " + str(e.message)
    return { "state": output }

File: sample_onsr/oci-objectstorage-onsr-put-object-ruby/func.rb
#
# oci-objectstorage-onsr-put-object-ruby version 1.0.
#
# Copyright (c) 2021 Oracle, Inc.  All rights reserved.
# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
#

require 'fdk'

# Certs are mounted at this location for ONSR realms
$cert_file_path = '/etc/ssl/cert.pem'
# file to upload
$file_to_upload = "onsr_cert_test"
$file_to_upload_content = "This is test file for ONSR test"

def put_object(bucket)
  FDK.log(entry: "put_object: enter")
  require 'oci'
  principal_signer =  OCI::Auth::Signers.resource_principals_signer
  object_storage_client = OCI::ObjectStorage::ObjectStorageClient.new(signer: principal_signer)
  namespace = object_storage_client.get_namespace.data

  object_storage_client.api_client.request_option_overrides = {
    ca_file: $cert_file_path
  }
  get_object_response = object_storage_client.put_object(namespace, bucket, $file_to_upload, $file_to_upload_content)
  FDK.log(entry: "put_object: exit")
end

def myfunction(context:, input:)
  bucket = input.respond_to?(:fetch) ? input.fetch('bucket') : input
  put_object(bucket)
  { message: "Completed Successfully!!!" }
end

FDK.handle(target: :myfunction)
